### PR TITLE
Added viz module with Colormap2D class for directional color encoding.

### PR DESCRIPTION
**Pull request by jan-car**
Signed-off-by: caron <j.caron@fz-juelich.de>

I reused the Colormap3D class from my "Pyramid" project, stripped away all superfluous content and simplified the functionality to work on 2D color encodings.

If you want, I'd like to add some testing, any hints how I should do that :-)?

I can also add the colour encoding at the appropriate locations, if you can point them out to me.

### CODE REVIEW

File: src/libertem/viz.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+    http://blog.asmartbear.com/color-wheels.html
+
+    """
+
+    _log = logging.getLogger(__name__ + '.ColormapHLS')
+
+    def __init__(self):
+        self._log.debug('Calling __init__')
+        h = np.linspace(0, 1, 256)
+        l = 0.5 * np.ones_like(h)
```
Comment (sk1p) [inline src/libertem/viz.py]: flake8 dislikes variable names that are ambiguous (`l` vs `I` etc.), thus the flake8 failure here. I don't feel strongly about this either way, we can either keep `l` as name and disable this check for this line with a `noqa` comment (see [flake8 documentation](http://flake8.pycqa.org/en/latest/user/violations.html#in-line-ignoring-errors) for details), or choose a different name.

Maybe for consistency disabling works best here?

File: src/libertem/viz.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 by Forschungszentrum Juelich GmbH
```
[the remark this answers is not in the document]
Comment (jan-car) [reply src/libertem/viz.py]: Not sure, I just use it in all my new files, same with the utf-8 encoding. I can remove this, if you like.

File: src/libertem/viz.py [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+
+import numpy as np
```
[the remark this answers is not in the document]
Comment (jan-car) [reply src/libertem/viz.py]: Makes sense, I usually have it the same way. Matplotlib and numpy are used so often, I nearly count them to the builtins :-P...

File: src/libertem/viz.py [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+from matplotlib import colors
+
+import colorsys
+
+import abc
+
+import logging
+
+
+__all__ = ['Colormap2D', 'ColormapCubehelix', 'ColormapPerception', 'ColormapHLS',
```
[the remark this answers is not in the document]
Comment (jan-car) [reply src/libertem/viz.py]: Yes, I erroneously removed them during decluttering...

File: src/libertem/viz.py (hunk 1 above)
Comment (jan-car) [reply src/libertem/viz.py]: I'll add the corresponding comment!

**Comment by jan-car** [comment]
Thanks for the feedback, I'm familiar with pytest and am using it myself, I just don't know what exactly I should test, this is usually a bit tricky for visualizations. I can test the `vector_to_rgb` function, to see if the colors are correct, but this would also depend on the chosen colorwheel ("cubehelix", "classic", etc.).
There's testing for matplotlib plots that check if the output is consistent pixel by pixel, but I'm not sure that's what you want here.
Any feedback on how I should test this without implementing concrete plotting commands, yet :-)?

File: src/libertem/viz.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sk1p) [reply src/libertem/viz.py]: Yes, please remove &copy;, UTF-8 encoding is fine.

**Comment by sk1p** [comment]
Thanks for the updates, looking good!

I think tests for `rgb_from_vector` using the cubehelix colorwheel would cover the most complex part of the code, right? Coverage as in 'doesn't crash if called' for the other colorwheels should be given by the initialization into `cmaps`.

Integration tests for the concrete implementation of the potting can be postponed until the integration is actually done.

Pixel-by-pixel tests aren't such a bad idea - are they easily implemented? Maybe they should also be done on a higher level, i.e. together with the integration tests.

**Comment by jan-car** [comment]
I added some tests for the two stand-alone functions. If coverage still complains, I might add some small tests for the different colormaps as well.

Concerning matplotlib testing, please have a look [here](https://matplotlib.org/devel/testing.html#writing-an-image-comparison-test)! I haven't done this myself, yet, but it looks easy enough. However, I thought you didn't want to rely on matplotlib as much, or did I misunderstand that?

**Comment by sk1p** [comment]
Thanks, merged! About visual tests, I'll defer this to later, and created https://github.com/LiberTEM/LiberTEM/issues/55 for this.

**Pull request merged**